src/adapter/threads.ts

```diff
@@ -382,6 +391,11 @@ export class Thread implements VariableStoreDelegate {
             scheduledPauseOnAsyncCall.id === event.asyncStackTraceId.id) {
           // Paused on the script which is run as a task for scheduled async call.
           // We are waiting for this pause, no need to resume.
+        } else if (
+          event.callFrames.length
+          && this.breakpoints.getBreakpointsAtLocation(event.callFrames[0].location).length
```

Comment:
I'd suggest this._handleSourceMapPause() to return boolean (whether to resume or not). It will end up in BreakpointManager anyway, which can make the decision. Otherwise, we start introducing more circular dependencies (BreakpointManager -> Thread -> BreakpointManager), which usually leads to less maintainable code. WDYT?

This way, it should pretty straightforward to check that any of uiLocations in Breakpoint.updateForSourceMap is at (0,0).